src/textual/command.py

```diff
@@ -68,43 +68,47 @@ class CommandSourceHit(NamedTuple):
     match_display: RenderableType
     """The Rich renderable representation of the hit.
 
-    Ideally a [rich Text object][rich.text.Text] object or similar.
+    Ideally a [rich Text object][rich.text.Text] object.
     """
 
-    command: CommandPaletteCallable
+    command: CallbackType
     """The function to call when the command is chosen."""
 
-    command_text: str
+    text: str | None = None
     """The command text associated with the hit, as plain text.
 
     This is the text that will be placed into the `Input` field of the
-    [command palette][textual.command_palette.CommandPalette] when a
+    [command palette][textual.command.CommandPalette] when a
     selection is made.
     """
 
-    command_help: str | None = None
+    help: str | None = None
     """Optional help text for the command."""
 
     def __lt__(self, other: object) -> bool:
-        if isinstance(other, CommandSourceHit):
-            return self.match_value < other.match_value
+        if isinstance(other, Hit):
+            return self.score < other.score
         return NotImplemented
 
     def __eq__(self, other: object) -> bool:
-        if isinstance(other, CommandSourceHit):
-            return self.match_value == other.match_value
+        if isinstance(other, Hit):
+            return self.score == other.score
         return NotImplemented
 
+    def __post_init__(self) -> None:
+        if self.text is None:
+            self.text = str(self.match_display)
 
-CommandMatches: TypeAlias = AsyncIterator[CommandSourceHit]
+
+Hits: TypeAlias = AsyncIterator[Hit]
 """Return type for the command source match searching method."""
 
 
-class CommandSource(ABC):
+class Source(ABC):
     """Base class for command palette command sources.
 
     To create a source of commands inherit from this class and implement
-    [`search`][textual.command_palette.CommandSource.search].
+    [`search`][textual.command.Source.search].
```

Comment:
```suggestion
    the method [`search`][textual.command.Source.search].
```

Reply:
"Implement the method search" ? Maybe "implement the search method", but think it reads ok as is.